public/requests.js

```diff
@@ -0,0 +1,111 @@
+import md5_encode from './MD5.js'
+
+/**
+ *  добавление eventListener на кнопку поиска трека
+ */
+const search_input = document.querySelector('.header__search');
+search_input.addEventListener('keydown', function(e) {
+    if (e.keyCode === 13) {
+        send_request('GET','track.search',search_input.value).then(
+            res => get_content('search',res.results.trackmatches.track)
+        )
+    }
+});
+
+/**
+ *  добавление eventListener на кнопку любимых треков
+ */
+const fav_button = document.querySelector('.fav');
+fav_button.addEventListener('click', function() {
+    send_request('GET','user.getlovedtracks').then(
+        res => get_content('favorites',res.lovedtracks.track)
+    )
+});
+
+/**
+ *
+ * @param method метод Апи (GET или POST)
+ * @param request_method параметр запроса - метод
+ * @param track название трека
+ * @param artist имя артиста
+ * @returns {Promise} Выполненный промис
+ */
+async function send_request(method,request_method,track= '',artist='') {
+    const url = 'http://ws.audioscrobbler.com/2.0/'
+    const api_key = '906db58ae0258689ba249d53210358ee';
+    if (method === 'GET') {
+        if (request_method === 'track.search') {
+            const response = await fetch(`${url}?api_key=${api_key}&method=${request_method}&track=${track}&format=json`, {
+                method: 'GET'
+            });
+            return await response.json()
+        }
+        else if (request_method === 'user.getlovedtracks') {
+            const response = await fetch(`${url}?api_key=${api_key}&method=${request_method}&user=Foxyb0y&format=json`, {
+                method: 'GET'
+            });
+            return await response.json();
+        }
+        else {
+            alert('Ошибка отправки запроса')
+        }
+    }
+    else if (method === 'POST') {
+        const sk = 'PEJSZfVNq8UlcfpQoiME1HzKEaFYAClf';
+        const secret = '04e5fe9b4835a8387149c2770345a9af';
+        const params = new URLSearchParams( {'track': track,'artist': artist,'api_key': api_key,
+            'api_sig': `${md5_encode(api_key,artist,sk,track,secret)}`,'sk': sk,'method': request_method});
+        const response = fetch(url, {
+            method: 'POST',
+            body: params,
+            headers: {'Content-Type':'application/x-www-form-urlencoded'}
+        })
+        return await response.json();
+    }
+}
+
+/**
+ *
+ * @param response_content определяет какой контент нужно получить
+ * @param track_array массив треков
+ * @param track название трека
+ * @param artist имя артиста
+ */
+function get_content(response_content,track_array,track=null,artist=null) {
+    const content = document.querySelector('.content');
+    content.innerHTML = "";
+    if (track_array.length !== 0) {
```

Comment:
Можно написать проще
```suggestion
    if (track_array.length) {
```